### PR TITLE
mgr/dashboard_v2: Configuration settings support

### PR DESCRIPTION
**Pull request by rjfd**
This PR adds the infrastructure to manage configuration settings in an easy way. For more information check the additions to the `README.rst` file.

This is also setting the ground for the implementation of the settings page.

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/README.rst [hunk 1]
```diff
@@ -204,3 +204,46 @@ case needs to add to the test.
...
 
+
+How to add a new configuration setting?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you need to store some configuration setting for a new feature, we already
+provide an easy mechanism for you to specify/use the new config setting.
+
+For instance, if you want to add a new configuration setting to hold the
+email address of the dashboard admin, just add a the setting name as a class
```
Comment (p-se) [inline src/pybind/mgr/dashboard_v2/README.rst]: superfluous `a` or `the`

**Review by p-se — APPROVED** [review]

File: src/pybind/mgr/dashboard_v2/README.rst (hunk 1 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/README.rst]: fixed

**Review by jecluis — APPROVED** [review]
Skimmed the code, looks okay. Getting my approval because I really, really like this idea. Being able to create config options in a simple way, without having to statically define them in multiple places AND even having the added benefit of the CLI getters and setters to be automatically available? <3
Also, very src/common/options.cc of you. Well done.

**Review by LenzGr — APPROVED** [review]
Nice work! Thanks.

**Review by votdev — APPROVED** [review]

**Comment by jcsp** [comment]
Is the intention to extend handle_option_command with the set_config/get_config calls so that this stores the config options?

**Comment by jecluis** [comment]
@jcsp do you mean by eventually setting these on config-key k/v pairs?

**Comment by jcsp** [comment]
I'm talking about MgrModule.get_config, MgrModule.set_config (the existing infrastructure for storing configuration)

**Comment by rjfd** [comment]
@jcsp the settings are already being stored in the config-key store. See: https://github.com/openattic/ceph/pull/46/files#diff-153cf734281d21281f908db208d1d068R36

**Comment by jcsp** [comment]
Ah - I missed that because it was the localized variants

You probably don't want to use localized configuration for everything by default: that would leave you having to set each mgr daemon's values separately.  Localized configuration is really more of a special case for settings that for some reason need to be different per mgr daemon, for example listen ports/addrs.

**Comment by rjfd** [comment]
@jcsp ah! didn't know that. Ok, I'll change it to the the "global" variant of get/set config.

**Comment by rjfd** [comment]
@jcsp does `set_config` stores the config in a namespace private to this mgr module, or the keys might conflict with keys of other mgr modules? If it's the latter, maybe I should add some prefix to the setting name to avoid conflicts...

**Comment by rjfd** [comment]
@jcsp nevermind, just found out that the keys are stored as `mgr/mgr_module_name/key`

**Review by sebastian-philipp — APPROVED** [review]
lgtm!

**Pull request merged**